### PR TITLE
add table to rebber and caption

### PR DESCRIPTION
**Pull request by artragis**
I still need :

- [ ] rebber `gridTables`
- [ ] rebber `table` and `gridTable` captions.

This is linked to @pierre-24 work on tables inside the latex class.

### CODE REVIEW

**Comment by vhf** [comment]
Please don't spend too much time on the rebber aspects of (grid)tables.

Priority goes to captions and other Markdown aspects. The LaTeX team hasn't decided how to handle tables yet: https://github.com/zestedesavoir/latex-template/issues/52

**Comment by artragis** [comment]
So, if rebber is not what we need, everything is ok for now : captions are tested and working

File: packages/remark-captions/src/index.js [hunk 1]
```diff
@@ -8,7 +8,10 @@ const legendBlock = {
 
 function plugin (opts) {
   if (opts) {
-    Object.keys(opts).map(key => legendBlock[key] = opts[key])
+    // not sure but I think I once saw a kind of "extend" function
+    Object.keys(opts).forEach(function (key) {
```
Comment (vhf) [inline packages/remark-captions/src/index.js]: https://github.com/zestedesavoir/zmarkdown/blob/ed65ab653321d3ee3a81316e2e0d9fe729f0b116/packages/rebber/src/escaper.js#L3-L26 😉

File: packages/rebber/src/types/table.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/* Expose. */
+const one = require('../one')
+module.exports = table
+
+const defaultMacro = (ctx, node) => {
+  let j = 0
+  const parsed = []
+  node.children.map(n => parsed.push(one(ctx, n, j++, node)))
```
Comment (vhf) [inline packages/rebber/src/types/table.js]: I think you're looking for `const parsed = node.children.map(n => one(ctx, n, j++, node))`

File: packages/rebber/src/types/table.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const inner = parsed.join('')
+  const lengths = []
+  inner.split('\\hline\n').map(l => lengths.push(l.split('&').length))
```
Comment (vhf) [inline packages/rebber/src/types/table.js]: Same here :)

File: packages/rebber/src/types/table.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  const cols = lengths.sort(cmp)[0]
+  let colHeader = '|'
+  let i = 0
+  for (;i < cols; i++) {
+    colHeader += 'c|'
+  }
```
Comment (vhf) [inline packages/rebber/src/types/table.js]: What about `String.prototype.repeat`?

File: packages/rebber/src/types/tableRow.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+/* Expose. */
+const one = require('../one')
+module.exports = tableRow
+
+const defaultMacro = (ctx, node) => {
+  let i = 0
+  const parsed = []
+  node.children.map(n => parsed.push(one(ctx, n, i++, node)))
```
Comment (vhf) [inline packages/rebber/src/types/tableRow.js]: What's the difference between this and `all(ctx, node)` ?

File: packages/remark-captions/README.md [hunk 6]
```diff
@@ -8,3 +8,54 @@ Source: A little green man, with a saber larger than himself
...
+    <tbody>
+      <tr>
+        <td>bla</td>
+        <td>bla</td>
+      </tr>
+    </tbody>
+  </table>
+  <figcaption>figcapt1</figcaption>
+</figure>
+```
```
Comment (vhf) [inline packages/remark-captions/README.md]: 😍

File: packages/zmarkdown/index.js [hunk 7]
```diff
@@ -62,7 +62,7 @@ const zmdParser = (config) => {
     .use(remarkHeadingShifter, config.headingShifter)
     .use(remarkIframes, config.iframes)
     .use(remarkGridTables)
-    .use(remarkCaptions)
+    .use(remarkCaptions, config.captions)
```
Comment (vhf) [inline packages/zmarkdown/index.js]: 👍

File: packages/rebber/src/types/table.js (hunk 2 above)
Comment (vhf) [inline packages/rebber/src/types/table.js]: or  https://github.com/zestedesavoir/zmarkdown/pull/82#discussion_r123068777 ?

File: packages/remark-captions/src/index.js (hunk 1 above)
Comment (artragis) [inline packages/remark-captions/src/index.js]: thks, will get it tomorrow

File: packages/rebber/src/types/table.js (hunk 2 above)
Comment (artragis) [inline packages/rebber/src/types/table.js]: in fact I tried this but it didn't work. However in between i made mor changements so perhaps I can rollback to this.

File: packages/rebber/src/types/table.js (hunk 4 above)
Comment (artragis) [inline packages/rebber/src/types/table.js]: is that really exist?

File: packages/rebber/src/types/tableRow.js (hunk 5 above)
Comment (artragis) [inline packages/rebber/src/types/tableRow.js]: all stringifies everything, here, I have an array of stringified

File: packages/remark-captions/README.md (hunk 6 above)
Comment (artragis) [inline packages/remark-captions/README.md]: yep, I thought of your pain to write all those readmes

File: packages/rebber/src/types/table.js (hunk 4 above)
Comment (vhf) [inline packages/rebber/src/types/table.js]: [String.prototype.repeat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat) 💃

File: packages/rebber/src/types/table.js (hunk 2 above)
Comment (vhf) [inline packages/rebber/src/types/table.js]: I see. By the way if you need an index you won't reuse later on, remember that `map`'s callback is `(item, index, items) => {}`.

Which means you could drop the `j` in the parent scope and do something like: const parsed = node.children.map((n, j) => one(ctx, n, j, node))`

**Pull request merged**